scripts/utils/termux_reuse_pr_build_artifacts.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+set -euo pipefail
+
+# Check if we can perform CI fast path
+# Here we make sure all commits come from single PR, make sure nor buildsystem nor dependencies were not changed
+# and if so we are reusing PR check artifacts and uploading them to apt repo to make save some CI time
+
+
+# Make sure we are running in a push triggered by PR merge.
+[[ "${GITHUB_EVENT_NAME:-}" == "push" ]] || exit 0
+
+error_exit() {
+  echo "$@"
+  exit 0
+}
+
+# OLD_COMMIT and HEAD_COMMIT are required
+[ -n "${OLD_COMMIT:-}" ] || error_exit "OLD_COMMIT is unset, not performing CI fast path"
+[ -n "${HEAD_COMMIT:-}" ] || error_exit "HEAD_COMMIT is unset, not performing CI fast path"
+[ -n "${GITHUB_TOKEN:-}" ] || error_exit "GITHUB_TOKEN is unset, not performing CI fast path"
+
+TERMUX_PACKAGE_DIRECTORIES="$(jq --raw-output 'del(.pkg_format) | keys | .[]' repo.json)"
+
+# We should obtain data about all commits in this push and make sure they all came from the same PR if any.
+# No hard tabs or newlines allowed inside a graphql request.
+GRAPHQL_QUERY=$(tr '\t\n' '  ' <<< "
+query {
+  repository(owner: \\\"termux\\\", name: \\\"termux-packages\\\") {
+  $(
+    for i in $(git rev-list --no-merges "$OLD_COMMIT..$HEAD_COMMIT" || true); do
+      echo "_${i:0:7}: object(oid: \\\"${i}\\\") { ...commitWithPR }"
+    done
+  )
+  }
+}
+
+fragment commitWithPR on Commit {
+  associatedPullRequests(first: 1) { nodes { baseRefOid headRefOid } edges { node { number } } }
+}")
+
+# Send graphql to github and obtain response
```

Comment:
GitHub's syntax parser gets really unhappy about the script at this point.
Probably due to the embedded GraphQL query above.

Reply:
You mean bash syntax parser or graphql explorer?

Reply:
I am not sure why graphql explorer could be unhappy. I replaced all `\"` with `"` in my query, pasted it to explorer and it worked fine with no warnings.

Reply:
I meant the syntax highlighting itself.
It thinks all of this below is a string.
Doesn't exactly matter as long as it works.